pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -297,13 +305,31 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 
 	if isOpenShift4 {
-		// Check whether client certificate secret exists
+		//Check whether client certificate secret exists
 		_, err := r.coreClient.Secrets(request.Namespace).Get(ctx, request.Name+"-tls-proxying", metav1.GetOptions{})
+		//TODO add automated regeneration of certificate after the expiration
 		if err != nil {
 			if errors.IsNotFound(err) {
-				reqLogger.Info("Client certificate secret must be created", "secret", request.Name+"-tls-proxying")
-				// Let's poll for the client certificate secret to be created
-				return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+				secretName := request.Name + "-tls-proxying"
+				reqLogger.Info("Client certificate not found, certificate will be created now", "secret", secretName)
+
+				caSecret, err := r.coreClient.Secrets("openshift-service-ca").Get(ctx, "signing-key", metav1.GetOptions{})
+				if err != nil {
+					reqLogger.Error(err, "Server certificate not found", err)
```

Comment:
I'd suggest to rephrase to `Reading certificate authority signing key failed`, as the error may not always _not found_.